LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan/DevNonce.cs

```diff
@@ -28,7 +27,29 @@ namespace LoRaWan
         public ushort AsUInt16 => this.value;
 #pragma warning restore IDE0032 // Use auto property
 
-        public override string ToString() => this.value.ToString(CultureInfo.InvariantCulture);
+        public override string ToString() => this.value.ToString(null, null);
+
+        public string ToString(string? format, IFormatProvider? formatProvider)
+        {
+#pragma warning disable format
+            return format switch
+            {
+                "N"                                 => ToHex(LetterCase.Upper),
+                "n"                                 => ToHex(LetterCase.Lower),
```

Comment:
`X` and `x` would be more standard here, but perhaps we don't need it at all? I don't think we should be treating dev nonce values in hexadecimal notation assuming they are just monotonically increasing numbers?

Reply:
I will remove the hex representation entirely

Reply:
Done in 9f3eb36e738fbaa1a3f376ab154540aa659eb33c